src/components/Context/index.js

```diff
@@ -17,28 +17,34 @@ import injectPlaceContext from "./injectPlaceContext";
 import injectTimestamp from "./injectTimestamp";
 import implementationDetails from "./implementationDetails";
 import createComponent from "./createComponent";
+import injectHighEntropyUserAgentHints from "./injectHighEntropyUserAgentHints";
 import { arrayOf, string } from "../../utils/validation";
 
 const web = injectWeb(window);
 const device = injectDevice(window);
 const environment = injectEnvironment(window);
 const placeContext = injectPlaceContext(() => new Date());
 const timestamp = injectTimestamp(() => new Date());
+const highEntropyUserAgentHints = injectHighEntropyUserAgentHints(navigator);
 
-const optionalContexts = {
+const defaultContexts = {
   web,
   device,
   environment,
   placeContext
 };
+const optionalContexts = {
+  highEntropyUserAgentHints,
```

Comment:
I probably missed the discussion about what to name this. We've talked about moving toward a system where you specify the specific fields in XDM you want populated, but that migration has been put on the back burner. That said, currently all the contexts are the top level properties in the xdm. For consistency, I'd prefer to see this context called "environment.browserDetails". Then in documentation explain that by default this is off, and when enabled may eat into the privacy budget.

Reply:
I agree that we need to mirror the XDM schema to get better granularity support on how to turn things on or off. `browserdetails` has a lot more than high-entropy user-agents in it though. It also contains `user-agent-client-hints` that holds low-entropy hints in addition to the high-entropy hints. Are you saying that you'd like to have different contexts represented by the full path to each object?

Reply:
I think "highEntropyUserAgentHints" is good for now. It matches AppMeasurement's naming, and is precise in what it does. Later if we move to using more granular XDM paths, we can use XDM paths for the client hints.